### PR TITLE
Fix: `nextToken` value in `logs:FilterLogEvents` response

### LINKED ISSUE #3882: logs client nextToken failure in filter_log_events
Similar to the problem reported in #3395 for `describe_log_groups`, `filter_log_events` shows the same errors when you try to use the `nextToken` to page through the events.

I'm working on a PR that uses a very similar fix to the one @bpandola implemented in #3398

### PR DESCRIPTION
**Pull request by neil-greenwood**
Plagiarizing freely from @bpandola and his PR #3398, I have
modified the pagination for FilterLogEvents to more closely follow
the real AWS behaviour.

Fixes #3882

### CODE REVIEW

File: moto/logs/models.py [hunk 1]
```diff
@@ -366,13 +366,35 @@ def filter_log_events(
...
             events = sorted(events, key=lambda event: event["timestamp"])
 
-        if next_token is None:
-            next_token = 0
+        first_index = 0
+        if next_token:
+            try:
+                group, stream, event_id = next_token.split("/")
+                if group != log_group_name:
+                    raise ValueError()
```
Comment (bblommers) [inline moto/logs/models.py:374]: Hi @neil-greenwood, thanks for the PR! Can we have a test for this error scenario as well?

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi

File: moto/logs/models.py (hunk 1 above)
Comment (neil-greenwood) [reply moto/logs/models.py:374]: Sure.

**Comment by neil-greenwood** [comment]
How important is the python2.7 compatibility? I'm struggling to get the timestamp working. I guess I'm being too fancy, and I should just use `int(time.time())` instead of bothering with the `datetime` library.

**Comment by bblommers** [comment]
> How important is the python2.7 compatibility?

Very - as long as we support it :) `time.time()` should definitely work, or alternatively `datetime.now()`

**Review by bblommers — APPROVED** [review]
LGTM!

**Pull request merged**